File: src/bin/http_server.rs
use actix_web::{web, App, HttpResponse, HttpServer};
use std::env;
use dotenv::dotenv;
use nats::asynk::Connection;


extern crate stock_broker_application;

use stock_broker_application::handlers::place_trade::create_trade_request;
use stock_broker_application::handlers::historical_data::get_historical_data;
use stock_broker_application::services::nats_service::connect_to_nats;

#[actix_web::main]
async fn main() -> std::io::Result<()> {
    dotenv().ok();

    let nats_url = env::var("NATS_URL").expect("NATS_URL must be set");
    let nats = connect_to_nats(&nats_url).await.expect("Failed to connect to NATS");

    let port = env::var("PORT").unwrap_or_else(|_| "8080".to_string());
    let port: u16 = port.parse().expect("PORT must be a number");

    HttpServer::new(move || {
        App::new()
            .data(nats.clone())
            .route("/trade", web::post().to(create_trade_request))
            .route("/historical_data", web::get().to(get_historical_data))
    })
    .bind(("0.0.0.0", port))?
    .run()
    .await
}

File: Go/bin/http_server.go
package main

import (
    "github.com/gin-gonic/gin"
    "github.com/joho/godotenv"
    "github.com/nats-io/nats.go"
    "log"
    "os"
    "stock_broker/handlers"
    "stock_broker/services"
)

func main() {
    err := godotenv.Load()
    if err != nil {
        log.Fatalf("Error loading .env file")
    }

    natsURL := os.Getenv("NATS_URL")
    nc, err := services.ConnectToNATS(natsURL)
    if err != nil {
        log.Fatalf("Failed to connect to NATS: %v", err)
    }

    port := os.Getenv("PORT")
    if port == "" {
        port = "8080"
    }

    router := gin.Default()
    router.POST("/trade", handlers.CreateTradeRequest(nc))
    router.GET("/historical_data", handlers.GetHistoricalData(nc))

    router.Run(":" + port)
}













































